example/src/tests/groupTests.ts

```diff
@@ -205,11 +207,68 @@ test('production MLS V3 client creation throws error', async () => {
   )
 })
 
+test('group message delivery status', async () => {
+  const [alixClient, boClient] = await createClients(2)
+  const alixGroup = await alixClient.conversations.newGroup([boClient.address])
+
+  await alixGroup.send('hello, world')
+
+  const alixMessages: DecodedMessage[] = await alixGroup.messages(true)
+
+  assert(
+    alixMessages.length === 2,
+    `the messages length should be 2 but was ${alixMessages.length}`
+  )
+
+  if (alixMessages[0].content() !== 'hello, world') {
+    throw new Error("newest message should be 'hello, world'")
+  }
+
+  assert(
+    alixMessages[0].deliveryStatus === 'UNPUBLISHED',
+    `the message should have a delivery status of ${MessageDeliveryStatus.UNPUBLISHED} but was ${alixMessages[0].deliveryStatus}`
+  )
+
+  const alexMessagesFiltered: DecodedMessage[] = await alixGroup.messages(
+    true,
+    undefined,
+    undefined,
+    undefined,
+    undefined,
+    MessageDeliveryStatus.UNPUBLISHED
+  )
```

Comment:
@alexrisch @cameronvoell this feels kind of bad for just wanting to set the delivery status? Should we be using a object like `messageOptions` instead of adding fields? This feels like how Swift and Kotlin work but doesn't seem to work out the same for typescript.